src/libmeasurement_kit/dns/getaddrinfo_async.hpp

```diff
@@ -106,11 +106,7 @@ void getaddrinfo_async(std::string name, addrinfo hints, SharedPtr<Reactor> reac
                       name.c_str(), error.code, error.what());
         std::vector<Answer> answers;
         if (!error && rp != nullptr) {
-            try {
-                answers = getaddrinfo_async_parse_response<inet_ntop>(name, rp);
-            } catch (const Error &err) {
-                error = std::move(err);
-            }
+            answers = getaddrinfo_async_parse_response<inet_ntop>(name, rp);
```

Comment:
This is probably the hunk that needs most attention. One needs to make sure that `getaddrinfo_async_parse_response` is not `throw`ing anymore.